mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -1553,51 +1565,79 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
           targetGemmDim1Attr.push_back(
               b.getNamedAttr("upper_layer_names",
                              b.getArrayAttr({b.getStringAttr(gemmKPad_name)})));
-        } else if (arg0TargetLayoutName2 == "gemmK") {
+        }
+        // filter of forward, gemmK=c*y*x
+        filterOobCheckDims.insert(nameToDims["c"]);
+        filterOobCheckDims.insert(nameToDims["y"]);
+        filterOobCheckDims.insert(nameToDims["x"]);
+      }
+
+      if (filterCheckPadGemmM) {
+        if (arg0TargetLayoutName1 == "gemmM") {
+          // wrw
           isFilterPad = true;
-          isGemmDim2Pad = true;
-          gemmDim2TargetName = b.getStringAttr(gemmKPad_name);
+          isGemmDim1Pad = true;
+          gemmDim1TargetName = b.getStringAttr(gemmMPad_name);
+          // even dim1 name is gemmM ,the size of dim 2 is gemmM
+          paddingFilterShape[2] = gemmMSize + gemmMExtra;
+
+          sourceGemmDim1Attr.push_back(
+              b.getNamedAttr("transformation", b.getStringAttr("Pad")));
+          sourceGemmDim1Attr.push_back(
+              b.getNamedAttr("parameters", b.getArrayAttr({
+                                               b.getI32IntegerAttr(0),
+                                               b.getI32IntegerAttr(gemmMExtra),
+                                           })));
 
-          paddingFilterShape[2] = paddingFilterShape[2] + gemmKExtra;
+          targetGemmDim1Attr.push_back(b.getNamedAttr(
+              "names", b.getArrayAttr({b.getStringAttr(gemmMPad_name)})));
+        } else if (arg0TargetLayoutName2 == "gemmM") {
+          // fwd
+          isFilterPad = true;
+          isGemmDim2Pad = true;
+          gemmDim2TargetName = b.getStringAttr(gemmMPad_name);
+          paddingFilterShape[2] = gemmMSize + gemmMExtra;
```

Comment:
comment for this line is needed.

Reply:
comment is not added. please refer to logic above where comments were supplied for `paddingFilterShape[2]`